<launch>
  <arg name="cam_width" default="1280" />
  <arg name="cam_height" default="960" />
  <arg name="cam_model" default="daA1280-54uc" />
  <arg name="camera_frame" default="camera" />

  <!-- Choose the calibration file that corresponds to the resolution being used. -->
  <arg name="cam_calibration_file" default="file://$(find basler_camera)/config/$(arg cam_model)_$(arg cam_width)x$(arg cam_height).yaml" />

  <node name="basler_camera" pkg="basler_camera" type="basler_camera_node" output="screen">
    <param name="frame_rate" type="int" value="10" />
    <param name="frame_id" type="string" value="$(arg camera_frame)" />
    <param name="camera_info_url" type="string" value="$(arg cam_calibration_file)" />
    <rosparam command="load" file="$(find basler_camera)/config/basler_params.yaml"/>
  </node>

  <node ns="camera" pkg="image_proc" type="image_proc" name="image_proc"/>
</launch>
